specification/hardwaresecuritymodules/resource-manager/Microsoft.HardwareSecurityModules/stable/2024-06-30/dedicatedhsm.json

```diff
@@ -0,0 +1,762 @@
+      "type": "string",
+      "description": "The type of identity.",
+      "enum": [
+        "User",
+        "Application",
+        "ManagedIdentity",
+        "Key"
+      ],
+      "x-ms-enum": {
+        "name": "identityType",
+        "modelAsString": true
+      }
+    },
+    "OutboundEnvironmentEndpointCollection": {
+      "description": "Collection of OutboundEnvironmentEndpoint",
+      "required": [
+        "value"
+      ],
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Collection of resources.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OutboundEnvironmentEndpoint"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "description": "Link to next page of resources.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "OutboundEnvironmentEndpoint": {
+      "description": "Egress endpoints which dedicated hsm service connects to for common purpose.",
+      "type": "object",
+      "properties": {
+        "category": {
+          "description": "The category of endpoints accessed by the dedicated hsm service, e.g. azure-resource-management, apiserver, etc.",
+          "type": "string"
+        },
+        "endpoints": {
+          "description": "The endpoints that dedicated hsm service connects to",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDependency"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "EndpointDependency": {
+      "description": "A domain name that dedicated hsm services are reaching at.",
+      "type": "object",
+      "properties": {
+        "domainName": {
+          "description": "The domain name of the dependency.",
+          "type": "string"
+        },
+        "endpointDetails": {
+          "description": "The Ports and Protocols used when connecting to domainName.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EndpointDetail"
+          },
+          "x-ms-identifiers": []
+        }
+      }
+    },
+    "EndpointDetail": {
+      "description": "Connect information from the dedicated hsm service to a single endpoint.",
+      "type": "object",
+      "properties": {
+        "ipAddress": {
+          "description": "An IP Address that Domain Name currently resolves to.",
+          "type": "string"
+        },
+        "port": {
+          "format": "int32",
+          "description": "The port an endpoint is connected to.",
+          "type": "integer"
+        },
+        "protocol": {
+          "description": "The protocol used for connection",
+          "type": "string"
+        },
+        "description": {
+          "description": "Description of the detail",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "DedicatedHsmName": {
+      "name": "name",
```

Comment:
>name [](http://example.com/codeflow?start=15&length=4)

Consider something a little more specific, e.g. hsmName.

Reply:
updated to dedicatedHsmName